appmap/unittest.py

```diff
@@ -12,23 +13,29 @@ def get_test_location(cls, method_name):
     return get_function_location(fn)
 
 
-def run(self, result=None):
-    method_name = self.id().split('.')[-1]
-    # Use the result's location if provided (e.g. by pytest),
-    # otherwise cobble one together ourselves.
-    if hasattr(result, 'location'):
-        location = result.location
-    else:
-        location = get_test_location(self.__class__, method_name)
+# unittest.case._Outcom.testPartExecutor is used by all supported
+# versions of unittest to run test cases. `isTest` will be True when
+# the part is the actual test method, False when it's setUp or
+# teardown.
+@wrapt.patch_function_wrapper('unittest.case', '_Outcome.testPartExecutor')
+@contextmanager
+def testPartExecutor(wrapped, _, args, kwargs):
+    def _args(test_case, *_, isTest=False, **__):
+        return (test_case, isTest)
+    test_case, is_test = _args(*args, **kwargs)
+    if not is_test:
+        with wrapped(*args, **kwargs):
```

Comment:
Is there a way to check or assert that the test isn't already driven by pytest? In case the user ends up somehow importing appmap.unittest.

Reply:
I agree that this would be good to do. Let me see if I can find something.

Reply:
The changes in this commit address this concern: `test_simple.py` now imports `appmap.unittest`, and the test cases confirm that the results are correct when driven by `pytest`, `appmap.unittest`, as well as `unittest`.

Reply:
The changes in da92d08b243a40ab2033233a4fa9dcf3a30221f6 address this concern: `test_simple.py` now imports `appmap.unittest`, and the test cases confirm that the results are correct when driven by `pytest`, `appmap.unittest`, as well as `unittest`.